nectarchain/calibration/container/waveforms.py

```diff
@@ -0,0 +1,350 @@
+    def __new__(cls,*args,**kwargs) : 
+        obj = object.__new__(cls)
+        return obj
+
+    def __init__(self,run_number : int,max_events : int = None,nevents : int = -1,run_file = None):
+        """construtor
+
+        Args:
+            run_number (int): id of the run to be loaded
+            maxevents (int, optional): max of events to be loaded. Defaults to 0, to load everythings.
+            nevents (int, optional) : number of events in run if known (parameter used to save computing time)
+            merge_file (optional) : if True will load all fits.fz files of the run, else merge_file can be integer to select the run fits.fz file according to this number
+            
+        """
+
+        self.__run_number = run_number
+        #gerer ici le fait de traiter plusieurs fichiers ou simplement 1 par 1
+        self.__reader = WaveformsContainer.load_run(run_number,max_events,run_file = run_file)
+
+        #from reader members
+        self.__npixels = self.__reader.camera_config.num_pixels
+        self.__nsamples =  self.__reader.camera_config.num_samples
+        self.__geometry = self.__reader.subarray.tel[WaveformsContainer.TEL_ID].camera
+        self.__subarray =  self.__reader.subarray
+        self.__pixels_id = self.__reader.camera_config.expected_pixels_id
+        
+        #set camera properties
+        log.info(f"N pixels : {self.npixels}")
+
+        #run properties
+        if nevents != -1 :
+            self.__nevents = nevents if max_events is None else min(max_events,nevents) #self.check_events()
+        else :
+            self.__nevents = self.check_events()
+            #reload file (bc check_events has drained reader generator)
+            self.__reader = WaveformsContainer.load_run(run_number,max_events,run_file = run_file)
+        log.info(f"N_events : {self.nevents}")
+
+        
+        
+
+        
+
+
+        #define zeros members which will be filled therafter
+        self.wfs_hg = np.zeros((self.nevents,self.npixels,self.nsamples),dtype = np.uint16)
+        self.wfs_lg = np.zeros((self.nevents,self.npixels,self.nsamples),dtype = np.uint16)
+        self.ucts_timestamp = np.zeros((self.nevents),dtype = np.uint64)
+        self.ucts_busy_counter = np.zeros((self.nevents),dtype = np.uint32)
+        self.ucts_event_counter = np.zeros((self.nevents),dtype = np.uint32)
+        self.event_type = np.zeros((self.nevents),dtype = np.uint8)
+        self.event_id = np.zeros((self.nevents),dtype = np.uint32)
+        self.trig_pattern_all = np.zeros((self.nevents,self.CAMERA.n_pixels,4),dtype = bool)
+        #self.trig_pattern = np.zeros((self.nevents,self.npixels),dtype = bool)
+        #self.multiplicity = np.zeros((self.nevents,self.npixels),dtype = np.uint16)
+
+
+    @staticmethod
+    def load_run(run_number : int,max_events : int = None, run_file = None) : 
+        """Static method to load from $NECTARCAMDATA directory data for specified run with max_events
```

Comment:
Is it documented somewhere that a `$NECTARCAMDATA` environment variable needs to be set ?

Reply:
I forgot to document that, is it fine if I explain in the README.md of nectarchain how to set this environment variable ? I'm not sure of the best policy

Reply:
For the time being, it may not be wise to document it at the module root level. I think for instance the `dqm` uses another environment variable for the same purpose. Or just leave it as it, with an explanation in the docstring.

Reply:
Ok I see, I will follow the dqm method